### PR TITLE
[Hieu] New functions in API system (in-depth check for images submission)

### PR DESCRIPTION
**Pull request by maybehieu**
Please review, currently quite junky code.

### CODE REVIEW

**Comment by maybehieu** [comment]
Incomplete

**Comment by maybehieu** [comment]
Re-opening for review.

File: Data_Collector_API/PreProcesser.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            return -3
+        elif blur == 1 and bright == -1:  # blurry and low-light
+            return -4
+        elif blur == 1 and bright == 0:  # blurry
+            return -2
+        elif blur == 0 and bright == 1:  # high-light
+            return -1
+        elif blur == 0 and bright == -1:  # low-light
+            return 0
+        return 1
```
Comment (pewdspie24) [inline Data_Collector_API/PreProcesser.py]: Make code DRY, too much elif

File: Data_Collector_API/PreProcesser.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    import glob
+    import natsort
+    import os
+    root = r"G:\CodeStuff\Rice_PTIT\new_data\\**\*"
+    files2 = glob.glob(root, recursive=True)
+    files2 = natsort.natsorted(files2, reverse=False)
+    blur = []
+    bright = []
+    ite = 0
+    processer = Processer()
```
Comment (pewdspie24) [inline Data_Collector_API/PreProcesser.py]: Processor, I think?

File: Data_Collector_API/main.py [hunk 3]
```diff
@@ -52,6 +74,38 @@ def predict():
...
+def get_img():
+    import io
+    import numpy as np
+    buffer = io.BytesIO()
+    file = request.files["file"]
+    byte_file = file.read()
+    file.save(buffer)
+    image = np.array(Image.open(file, formats=["JPEG"]))
+    image = Image.fromarray(image)
+    processer = PreProcesser.Processer(byte_file)
```
Comment (pewdspie24) [inline Data_Collector_API/main.py]: Same thing here

File: Data_Collector_API/main.py [hunk 4]
```diff
@@ -52,6 +74,38 @@ def predict():
...
+        response = make_response(json.dumps(result))
+        return response
+    else:
+        # save invalid image for later inspection
+        time_now = datetime.now().strftime("%Y%m%d_%H%M%S")
+        dir_path = os.path.join(str(0), time_now + ".jpg")
+        if os.path.exists(os.path.join(cwd, dir_path)):
+            time.sleep(1)
+            time_now = datetime.now().strftime("%Y%m%d_%H%M%S")
+            dir_path = os.path.join(str(0), time_now + ".jpg")
```
Comment (pewdspie24) [inline Data_Collector_API/main.py:82]: Is there a case that 2 or more images can be processed in 1 sec?

**Review by pewdspie24 — COMMENTED** [review]
Pls check

**Comment by maybehieu** [comment]
I just pushed some changes in attempt of fixing the errors you've listed above, can you review it again?

**Comment by pewdspie24** [comment]
LGTM

**Pull request merged**